plugins/modules/ipapermission.py

```diff
@@ -0,0 +1,523 @@
+
+                        # Add members
+                        if len(privilege_add) > 0:
+                            commands.append([name, "permission_add_member",
+                                             {
+                                                 "privilege": privilege_add,
+                                                 "no_members": no_members_value
+                                             }])
+                        # Remove members
+                        if len(privilege_del) > 0:
+                            commands.append([name, "permission_remove_member",
+                                             {
+                                                 "privilege": privilege_del,
+                                                 "no_members": no_members_value
+                                             }])
+                elif action == "member":
+                    if res_find is None:
+                        ansible_module.fail_json(
+                            msg="No permission '%s'" % name)
+
+                    if privilege is None:
+                        ansible_module.fail_json(msg="No privilege given")
+
+                    commands.append([name, "permission_add_member",
+                                     {
+                                         "privilege": privilege,
+                                         "no_members": no_members_value
+                                     }])
+                else:
+                    ansible_module.fail_json(
+                        msg="Unknown action '%s'" % action)
+            elif state == "renamed":
+                if action == "permission":
+                    # Generate args
+                    # Note: Only valid arg for rename is rename.
+                    args = gen_args(perm_rights, attrs, perm_bind_rule_type,
+                                    perm_location, extra_target_filter,
+                                    perm_target_filter, perm_target,
+                                    perm_target_to, perm_target_from,
+                                    member_of, target_group, object_type,
+                                    no_members, rename)
+
+                    # Found the permission
+                    if res_find is not None:
+                        # For all settings is args, check if there are
+                        # different settings in the find result.
+                        # If yes: modify
+                        if not compare_args_ipa(ansible_module, args,
+                                                res_find):
+                            commands.append([name, "permission_mod", args])
+                    else:
+                        ansible_module.fail_json(
+                            msg="Permission not found, cannot rename")
+                elif action == "member":
+                    ansible_module.fail_json(
+                        msg="Action type member, cannot rename")
```

Comment:
This check could be made when checking parameters, to fail early, and with a more "commom" error message for this type or error.

Reply:
I think I addressed this in the latest commits.